Compiler/src/ssair/show.jl

```diff
@@ -105,11 +105,18 @@ function print_stmt(io::IO, idx::Int, @nospecialize(stmt), code::Union{IRCode,Co
             printstyled(io, "dynamic invoke "; color = :yellow)
             abi = (ci::Core.MethodInstance).specTypes
         end
-        show_unquoted(io, stmt.args[2], indent)
-        print(io, "(")
         # XXX: this is wrong if `sig` is not a concretetype method
         # more correct would be to use `fieldtype(sig, i)`, but that would obscure / discard Varargs information in show
         sig = abi == Tuple ? Core.svec() : Base.unwrap_unionall(abi).parameters::Core.SimpleVector
+        ft = maybe_argextype(stmt.args[2], code, sptypes)
+        if stmt.args[2] isa GlobalRef && (singleton_type(ft) === singleton_type(sig[1]) !== nothing)
```

Comment:
I trust that you know what you're doing, but does the implication `stmt.args[1] isa GlobalRef` => `abi != Tuple` hold? It doesn't strike me that it is the case, so we may have an out of bounds access.